hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -969,10 +977,25 @@ void doDrain(final List<RAMQueueEntry> entries) throws InterruptedException {
           }
           index++;
         } catch (BucketAllocatorException fle) {
-          LOG.warn("Failed allocation for " + (re == null ? "" : re.getKey()) + "; " + fle);
+          long currTE = System.currentTimeMillis()/1000; // Current time since Epoch in seconds.
+          // Record the warning.
+          cacheStats.allocationFailed(re.getData().getSerializedLength());
+          if (prevRecTE == 0) {
```

Comment:
May be just a common log is enoug?
if( prevRecTE == 0 || currTE - prevRecTE > ALLOCATION_FAIL_TRACE_TIME_GAP)